rootston/desktop.c

```diff
@@ -107,14 +107,26 @@ bool view_center(struct roots_view *view) {
 	return true;
 }
 
-void view_initialize(struct roots_view *view) {
+void view_setup(struct roots_view *view) {
 	view_center(view);
-	struct roots_input *input = view->desktop->server->input;
 
+	struct roots_input *input = view->desktop->server->input;
 	set_view_focus(input, view->desktop, view);
 	wlr_seat_keyboard_notify_enter(input->wl_seat, view->wlr_surface);
 }
 
+void view_teardown(struct roots_view *view) {
```

Comment:
What if the view being torn down is not the focused view?

Reply:
That's taken care of with `views->items[views->length-1] != view`.